File: web/src/types/collection.ts
export interface CollectionSummary {
  id: string
  name: string
  recipe_count: number
  created_at: string
  contains_recipe?: boolean | null
}

export interface RecipeCollectionTag {
  id: string
  name: string
}

export interface CollectionListResponse {
  items: CollectionSummary[]
  total: number
}

export interface CollectionRecipeSummary {
  id: string
  title: string
  step_count: number
  used_ai: boolean
  created_at: string
}

export interface CollectionDetailResponse {
  id: string
  name: string
  created_at: string
  recipes: CollectionRecipeSummary[]
}

export interface ShareResponse {
  is_public: boolean
  share_slug?: string | null
  share_url?: string | null
}

export interface CommunityRecipeSummary {
  slug: string
  title: string
  author_name: string
  step_count: number
  used_ai: boolean
  created_at: string
  icon_url?: string | null
}

export interface CommunityRecipeListResponse {
  items: CommunityRecipeSummary[]
  total: number
}

export interface SharedRecipeResponse {
  slug: string
  title: string
  recipe: import('./recipe').ParsedRecipe
  author_name: string
  step_count: number
  used_ai: boolean
  source_type?: string
  source_url?: string | null
  icon_url?: string | null
}

File: backend/app/schemas/collection.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class CreateCollectionRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)


class UpdateCollectionRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)


class CollectionSummary(BaseModel):
    id: str
    name: str
    recipe_count: int
    created_at: str
    contains_recipe: Optional[bool] = None


class CollectionListResponse(BaseModel):
    items: list[CollectionSummary]
    total: int


class CollectionRecipeSummary(BaseModel):
    id: str
    title: str
    step_count: int
    used_ai: bool
    created_at: str


class CollectionDetailResponse(BaseModel):
    id: str
    name: str
    created_at: str
    recipes: list[CollectionRecipeSummary]


class RecipeCollectionTag(BaseModel):
    id: str
    name: str
